editors/sc-ide/core/settings/serialization.cpp

```diff
@@ -180,12 +172,12 @@ bool readSettings(QIODevice &device, QSettings::SettingsMap &map)
 
     try {
         boost::iostreams::stream<IODeviceSource> in( &device );
-        Parser parser(in);
-        Node doc;
-        while(parser.GetNextDocument(doc)) {
-            if( doc.Type() != NodeType::Map ) continue;
-            QString key;
-            parseNode( doc, key, map );
+        Node doc = Load( in );
+        if( doc ) {
+            if( doc.IsMap() ) {
+                QString key;
+                parseNode( doc, key, map );
+            }
```

Comment:
I'm pretty sure this `while`->`if` is causing the IDE to not load stored settings. Appears to be a no-op.